python-libraries/data-platform-catalogue/data_platform_catalogue/client/datahub/search.py

```diff
@@ -127,6 +133,41 @@ def search_facets(
         response = response["aggregateAcrossEntities"]
         return self._parse_facets(response.get("facets", []))
 
+    def list_data_product_assets(
+        self, urn: str, count: int, start: int = 0
+    ) -> SearchResponse:
+        """
+        returns a SearchResponse containing all assets in given data product (by urn)
+        """
+        variables = {
+            "urn": urn,
+            "query": "*",
+            "start": start,
+            "count": count,
+        }
+        try:
+            response = self.graph.execute_graphql(
+                self.data_product_asset_query, variables
+            )
+        except GraphError as e:
+            raise Exception("Unable to execute listDataProductAssets query") from e
+        page_results = []
+        for result in response["listDataProductAssets"]["searchResults"]:
+            entity = result["entity"]
+            entity_type = entity["type"]
+            matched_fields = {
+                i["name"]: i["value"] for i in result.get("matchedFields", [])
+            }
+            if entity_type == "DATASET":
+                page_results.append(self._parse_dataset(entity, matched_fields))
+            else:
+                raise ValueError(f"Unexpected entity type: {entity_type}")
```

Comment:
Will we get different entity types coming back when we start adding charts to data products?

If so we might want to filter these out for now, rather than throw an error

Reply:
Yeah was wondering here as similar to search method. As we're not sure how to parse other result types yet i thought it be difficult to catch and could add other types in as and when we add new entity types to datahub.

Could be another ticket to make a more comprehensive `ResultType` enum and ensure our search client methods handle each type?